### PR TITLE
feat(server): Expose launchServer as api 

### PR DESCRIPTION
**Pull request by TheLarkInn**
* expose launchServer
* expose writePath so paths can be used by default or customized
* make VIZ_PATH an export for reuse

### CODE REVIEW

File: utils.ts [hunk 1]
```diff
@@ -67,6 +70,16 @@ export function sourceMapToLineHits(hitTracks: Map<string, SourceTrack>) {
   return sourceToLineMapping;
 }
 
+export function getWritePathForSerializedData(
+  visualizationPath?: string,
```
Comment (samccone) [inline utils.ts]: visualizationPath: string = VIZ_PATH

Comment (TheLarkInn) [inline utils.ts]: AHH I was wondering how the defaults work. Sweet.

File: index.ts [hunk 2]
```diff
@@ -67,7 +48,10 @@ if (cli.flags["demo"]) {
     console.log(stringifedData);
   } else {
     const dataPath = `data_${Date.now()}`;
-    fs.writeFileSync(path.join(__dirname, VIZ_PATH, dataPath), stringifedData);
-    launchServer(dataPath);
+    const writePath = getWritePathForSerializedData(dataPath);
+
+    fs.writeFileSync(writePath, stringifedData);
+
+    launchServer(dataPath, VIZ_PATH);
```
Comment (samccone) [inline index.ts]: args are flipped here I think

Comment (TheLarkInn) [inline index.ts]: Ah I flipped them in one place not another.

**Comment by samccone** [comment]
Let me know when this is ready for another pass.

Thanks!

**Comment by TheLarkInn** [comment]
Had to drop everything for work. Want your opinion though.

I'm not finding a decent way of getting absolute path to node module from the module itself. Would you think mounting the viz should be hidden in the module or rather somewhere else like the webpack users dist directory. Otherwise I can just add an optional param to pass the directory context to resolve the real module path.

File: utils.ts [hunk 3]
```diff
@@ -67,6 +70,22 @@ export function sourceMapToLineHits(hitTracks: Map<string, SourceTrack>) {
   return sourceToLineMapping;
 }
 
+/**
```
Comment (TheLarkInn) [inline utils.ts]: So because I've been struggling with how to name the argument context, I instead documented the exports in hopes to better explain it.

**Comment by TheLarkInn** [comment]
Alright @samccone let's let it rip. This felt like the _cleanest_ way. 

If there is an easy solution id be more than open to it. Just trying to really make the most out of reusing the server to an extent.

**Comment by samccone** [comment]
I had to revert this, as it broke launching the server.

if you run your change locally by

yarn run build
./bin.js --demo

note how the page does not render.

**Pull request merged**